source/SpinalHDL/Simulation/bootstraps.rst

```diff
@@ -79,18 +79,18 @@ This function will return a ``SimCompiled`` instance.
 
 On this ``SimCompiled`` instance you can run your simulation with the following functions:
 
-.. list-table::
-   :header-rows: 1
-
-   * - Syntax
-     - Description
-   * - ``doSim[(simName[, seed])]{dut => ...}``
-     - Run the simulation until the main thread is done (doesn't wait on forked threads) or until all threads are stuck
-   * - ``doSimUntilVoid[(simName[, seed])]{dut => ...}``
-     - Run the simulation until all threads are done or stuck
+``doSim[(simName[, seed])]{dut => /* main stimulus code */}``
+  Run the simulation until the main thread runs to completion and exits/returns.
+  It will detect and report an error if the simulation gets fully stuck. As long as
+  e.g. a clock is running the simulation can continue forever, it is therefore recommended
+  to use ``SimTimeout(cycles)`` to limit the possible runtime.
 
+``doSimUntilVoid[(simName[, seed])]{dut => ...}``
+  Run the simulation until it is ended by calling either ``simSuccess()`` or ``simFailure()``.
+  The main stimulus thread can continue or exit early, as long as there are events to process,
+  the simulation will continue. The simulation will report an error if it gets fully stuck.
```

Comment:
I think there is an issue with the code: wouldn't a function named `doSimUntilVoid` be expected to end successfully when void is reach / there are no events to process anymore?

Reply:
That was the initial intend, but it is kinda never used at all and not usefull, and instead provide some easy to produce false SimSuccess.
So we turned it into SimFailure if the simulation has nothing scheduled anymore.

Reply:
So the name did match the behavior, but it does not anymore… So the name does not match the behavior!